BLE GATT

JonaMata opened the pull request.

JonaMata committed:
Added BLE GATT search and connect support

JonaMata committed:
Added notify and data reading functionality, replaced pybluez with bleak

JonaMata committed:
Remove connection modes

JonaMata committed:
Start workiong on threading

FrankBosman committed:
Added a demo file that shows the data in a graph

JonaMata committed:
Add SensorManager and type hinting

JonaMata committed:
Update README with BLE implementation including API reference

JonaMata committed:
Fix List type hinting and update README with measurement data description

ttsmit commented on setup.py:
```diff
@@ -27,7 +27,6 @@
     python_requires='>=3.6',
     install_requires=[
         'pyserial >= 3.5',
```
As far as I see, the serial option is not necessarily any more, thus this requirement can be dropped.

ttsmit commented on examples/using_sensor_manager.py:
Works for now, would be nice to implement an extensive example like `read_sensor_demo` as well utilizing the Sensor Manager.

ttsmit commented on examples/using_sensor_manager.py:
```diff
@@ -0,0 +1,22 @@
+
+
+# Create a sensor manager for the given sensor names using the given callback
+manager = SensorManager(['0BE6'], callback)
```
Do you want to keep this sensor value in here?

ttsmit commented on examples/read_sensor.py:
I think better to remove this, it only creates confusion.

ttsmit commented on examples/read_sensor_demo.py:
I could not get this to work.

JonaMata committed:
Added emulator, sensor frimware version and sample rate feedback

- Added record method to SensorManager and SenesorEmulator to emulate sensors
- Add possibility to read firmware version from sensor
- Move delay calculation from python to firmware and add feedback of actual set sample rate

JonaMata committed:
Update docs

JonaMata committed:
Set minimum python version to 3.7

Frank-Bosman commented on docs/api.md:
```diff
@@ -0,0 +1,6 @@
+# Table of Contents
+
+* [CreaTeBME.SensorManager](./SensorEmulator.md)
+* [CreaTeBME.SensorEmulator](./SensorManager.md)
```
```suggestion
* [CreaTeBME.SensorEmulator](./SensorEmulator.md)
```

Frank-Bosman commented on docs/api.md:
```diff
@@ -0,0 +1,6 @@
+# Table of Contents
+
+* [CreaTeBME.SensorManager](./SensorEmulator.md)
```
```suggestion
* [CreaTeBME.SensorManager](./SensorManager.md)
```

Frank-Bosman commented on examples/read_sensor.py:
```diff
@@ -0,0 +1,41 @@
+
+
+    # while True:
+    #     await asyncio.sleep(1)
```
```suggestion

```
Old commented code

Frank-Bosman commented on examples/read_sensor.py:
```diff
@@ -0,0 +1,41 @@
+
+
+def print_measurement(data):
+    print('het werkt:', data)
```
```suggestion
    print('Sensor data:', data)
```
Change this to be English

Frank-Bosman commented on examples/read_sensor.py:
```diff
@@ -0,0 +1,41 @@
+        await sensor.set_callback(print_measurement)
+        await sensor.set_sample_rate(100)
+
+    print('yay')
```
```suggestion

```
Probably remove this

Frank-Bosman commented on examples/read_sensor_demo.py:
I'm just going to skip over this as i will create a new branch to night or tomorrow

Frank-Bosman reviewed (commented):
A few small changes, but apart from those  it looks good

JonaMata committed:
Update docs/api.md

Co-authored-by: Frank-Bosman <94541505+Frank-Bosman@users.noreply.github.com>

JonaMata committed:
Update docs/api.md

Co-authored-by: Frank-Bosman <94541505+Frank-Bosman@users.noreply.github.com>

JonaMata committed:
Update examples/read_sensor.py

Co-authored-by: Frank-Bosman <94541505+Frank-Bosman@users.noreply.github.com>

JonaMata committed:
Update examples/read_sensor.py

Co-authored-by: Frank-Bosman <94541505+Frank-Bosman@users.noreply.github.com>

JonaMata committed:
Update examples/read_sensor.py

Co-authored-by: Frank-Bosman <94541505+Frank-Bosman@users.noreply.github.com>

JonaMata committed:
Fix stop and start to disconnect and connect sensor

FrankBosman committed:
Added an is_running to the SensorManager.py and SensorEmulator.py

JonaMata committed:
Fix docstring

JonaMata committed:
Fix starting cancelled timer
